### PR TITLE
Add nitpicker, which allows adding comments to PRs when they match a file pattern

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #133870
* __->__ #133861

This message would have helped avoid https://www.internalfb.com/sevmanager/view/440895

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

File: .github/nitpicks.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+- markdown: |
+    ## Attention! native_functions.yaml was changed
+    If you are adding a new function or defaulted argument to native_functions.yaml, you cannot use it from pre-existing Python frontend code until our FC window passes (two weeks).  Split your PR into two PRs, one which adds the new C++ functionality, and one that makes use of it from Python, and land them two weeks apart.
```
Comment (albanD) [inline .github/nitpicks.yml:3]: nit
```suggestion
    ## Attention! native_functions.yaml was changed
    
    ### Be careful about Backward Compatibility
    If you are removing and modifying an existing function, make sure the backward compatibility window has been properly followed.
    See [here](https://github.com/pytorch/pytorch/wiki/PyTorch's-Python-Frontend-Backward-and-Forward-Compatibility-Policy) for more details.
    
    ### Be careful about Forward Compatibility
    If you are adding a new function or defaulted argument to native_functions.yaml, you cannot use it from pre-existing Python frontend code until our FC window passes (two weeks).  Split your PR into two PRs, one which adds the new C++ functionality, and one that makes use of it from Python, and land them two weeks apart.
    See [here](https://github.com/pytorch/pytorch/wiki/PyTorch's-Python-Frontend-Backward-and-Forward-Compatibility-Policy) for more details.
    
    ### Be careful about public APIs
    Any new function here that doesn't start with `_` or use a special namespace will be added to the python `torch.*` and/or `torch.Tensor.*` namespace by default. Make sure this is the intended behavior and the PR contains the appropriate docstring and docs website update for your new function/method.
```

**Review by albanD — APPROVED** [review]
Sounds good!
Only small proposal to add even more context

File: .github/workflows/nitpicker.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    runs-on: ubuntu-latest
+    # Do not comment on nightly builds PR
+    if: ${{ github.event.pull_request.number != 26921 && github.repository_owner == 'pytorch' }}
+    steps:
+    - name: Checkout PyTorch
+      uses: pytorch/pytorch/.github/actions/checkout-pytorch@main
+    - uses: ethanis/nitpicker@v1
+      with:
+        nitpicks: '.github/nitpicks.yml'
+        repo-token: "${{ secrets.GITHUB_TOKEN }}"
```
Comment (albanD) [inline .github/workflows/nitpicker.yml:26]: Should be "token" I think

**Review by izaitsevfb — APPROVED** [review]

**Comment by ezyang** [comment]
close open to trigger nitpicker job

**Comment by kit1980** [comment]
@ezyang I don't think works in non-ghstack PRs.
`pull_request_target` runs in the context of the base branch (main), so it will check the code from main, not PR.

**Comment by kit1980** [comment]
I don't understand how this works for non-ghstack PRs (and I verified it works).
For `pull_request_target`, the workflow runs in the context of the base branch (main), and you can see it actually checkouts main of PyTorch https://github.com/pytorch/pytorch/actions/runs/11026434358/job/30622960083?pr=136607
But how does ethanis/nitpicker@v1 get the code of the PR?

**Pull request closed without merging**